build/templates/session.py.mako

```diff
@@ -87,6 +87,20 @@ class ${session_context_manager}(object):
 
 
 % endif
+% if config['use_session_lock']:
+class _Lock(object):
+    def __init__(self, session):
+        self._session = session
+
+    def __enter__(self):
+        self._caller_has_lock = self._session.lock_session(caller_has_lock=False)
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self._caller_has_lock = self._session.unlock_session(self._caller_has_lock)
```

Comment:
If, as suggested, we get rid of the optional input parameter, then this becomes

``` python
if self._caller_has_lock:
    self._session._unlock_session()
```

Reply:
Why would we get rid of it? It still makes sense to allow using it for the same use cases as in the documentation.

Reply:
I need to read more on it.